File: src/background.lua
local Background = {}
Background.__index = Background

function Background.new(game)
	local self = setmetatable({}, Background)

	self.game = game
	self.x = 0
	self.offsetX = 0
	self.group = display.newGroup()
	self.group.y = -65
	self.scale = 2.2
	self.group.xScale = self.scale
	self.group.yScale = self.scale

	for j = 1, 2, 1 do
		local image = display.newImage(self.group, 'images/background.png')
		image.x = (j - 1) * image.width
		self.width = image.width
	end

	return self
end

function Background:update(delta)
	self.offsetX = self.offsetX + delta * -0.02 * self.game.speed

	while (self.offsetX <= -self.width) do
		self.offsetX = self.offsetX + self.width
	end

	self.group.x = self.offsetX * self.scale

end

return Background

File: src/background.js
define(['assets'], function (Assets) {
	function Background(game) {
		PIXI.DisplayObjectContainer.call(this);

		var j, sprites = [], sprite;
		this.game = game;
		this.x = 0;
		this.offsetX = 0;
		this.position.y = -40;
		this.scale.x = 2.2;
		this.scale.y = 2.2;

		for (j = 0; j < 2; j += 1) {
			sprite = new PIXI.Sprite(Assets.background);
			sprite.position.x = Math.floor(j * sprite.width);
			sprite.position.y = 0;
			sprites.push(sprite);
			this.width = sprite.width;
			this.addChild(sprite);
		}
	}

	Background.prototype = new PIXI.DisplayObjectContainer();

	Background.prototype.update = function (delta) {
		this.offsetX = this.offsetX + delta * -0.02 * this.game.speed;

		while (this.offsetX <= -this.width) {
			this.offsetX = this.offsetX + this.width;
		}

		this.position.x = Math.floor(-90 + this.offsetX * this.scale.x);
	};

	return Background;
});
